src/hotspot/share/opto/loopopts.cpp

```diff
@@ -1929,6 +1931,19 @@ bool PhaseIdealLoop::ctrl_of_all_uses_out_of_loop(const Node* n, Node* n_ctrl, I
   return true;
 }
 
+// Sinking a node from a pre loop to its main loop pins the node between the pre and main loops. If that node is input
+// to a check that's eliminated by range check elimination, it becomes input to an expression that feeds into the exit
+// test of the pre loop above the point in the graph where it's pinned.
```

Comment:
I guess the alternative would have been not to do that RC elimination, right?
If yes: you could finish the thought and say that we prefer to have a chance at RC elimination, rather than sinking the node out of the pre-loop.